This patch file is used by install.sh.

--- doc/misc/Makefile.in	2010-10-06 04:20:46 +0000
+++ doc/misc/Makefile.in	2010-10-08 05:27:56 +0000
@@ -55,6 +55,7 @@
 	$(infodir)/emacs-mime \
 	$(infodir)/epa \
 	$(infodir)/erc \
+	$(infodir)/ert \
 	$(infodir)/eshell \
 	$(infodir)/eudc \
 	$(infodir)/efaq \
@@ -105,6 +106,7 @@
 	emacs-mime.dvi \
 	epa.dvi \
 	erc.dvi \
+	ert.dvi \
 	eshell.dvi \
 	eudc.dvi \
 	faq.dvi \
@@ -155,6 +157,7 @@
 	emacs-mime.pdf \
 	epa.pdf \
 	erc.pdf \
+	ert.pdf \
 	eshell.pdf \
 	eudc.pdf \
 	faq.pdf \
@@ -342,6 +345,14 @@
 erc.pdf: erc.texi
 	$(ENVADD) $(TEXI2PDF) ${srcdir}/erc.texi
 
+ert : $(infodir)/ert
+$(infodir)/ert: ert.texi $(infodir)
+	cd $(srcdir); $(MAKEINFO) ert.texi
+ert.dvi: ert.texi
+	$(ENVADD) $(TEXI2DVI) ${srcdir}/ert.texi
+ert.pdf: ert.texi
+	$(ENVADD) $(TEXI2PDF) ${srcdir}/ert.texi
+
 eshell : $(infodir)/eshell
 $(infodir)/eshell: eshell.texi $(infodir)
 	cd $(srcdir); $(MAKEINFO) eshell.texi
--- doc/misc/makefile.w32-in	2010-01-13 08:35:10 +0000
+++ doc/misc/makefile.w32-in	2010-10-10 01:43:36 +0000
@@ -43,7 +43,8 @@
 		$(infodir)/org $(infodir)/url $(infodir)/speedbar \
 		$(infodir)/tramp $(infodir)/ses $(infodir)/smtpmail \
 		$(infodir)/flymake $(infodir)/newsticker $(infodir)/rcirc \
-		$(infodir)/erc $(infodir)/remember $(infodir)/nxml-mode \
+		$(infodir)/erc $(infodir)/ert \
+		$(infodir)/remember $(infodir)/nxml-mode \
 		$(infodir)/epa $(infodir)/mairix-el $(infodir)/sasl \
 		$(infodir)/auth $(infodir)/eieio $(infodir)/ede \
 		$(infodir)/semantic $(infodir)/edt
@@ -54,7 +55,8 @@
 		ada-mode.dvi autotype.dvi idlwave.dvi eudc.dvi ebrowse.dvi \
 		pcl-cvs.dvi woman.dvi eshell.dvi org.dvi url.dvi \
 		speedbar.dvi tramp.dvi ses.dvi smtpmail.dvi flymake.dvi \
-		newsticker.dvi rcirc.dvi erc.dvi remember.dvi nxml-mode.dvi \
+		newsticker.dvi rcirc.dvi erc.dvi ert.dvi \
+		remember.dvi nxml-mode.dvi \
 		epa.dvi mairix-el.dvi sasl.dvi auth.dvi eieio.dvi ede.dvi \
 		semantic.dvi edt.dvi
 INFOSOURCES = info.texi
@@ -301,6 +303,11 @@
 erc.dvi: erc.texi
 	$(ENVADD) $(TEXI2DVI) $(srcdir)/erc.texi
 
+$(infodir)/ert: ert.texi
+	$(MAKEINFO) ert.texi
+ert.dvi: ert.texi
+	$(ENVADD) $(TEXI2DVI) $(srcdir)/ert.texi
+
 $(infodir)/epa: epa.texi
 	$(MAKEINFO) epa.texi
 epa.dvi: epa.texi
@@ -358,7 +365,7 @@
 		 $(infodir)/url* $(infodir)/org* \
 		 $(infodir)/flymake* $(infodir)/newsticker* \
 		 $(infodir)/sieve* $(infodir)/pgg* \
-		 $(infodir)/erc* $(infodir)/rcirc* \
+		 $(infodir)/erc* $(infodir)/ert* $(infodir)/rcirc* \
 		 $(infodir)/remember* $(infodir)/nxml-mode* \
 		 $(infodir)/epa* $(infodir)/sasl* \
 		 $(infodir)/mairix-el* $(infodir)/auth* \
--- configure.in	2010-10-03 15:39:21 +0000
+++ configure.in	2010-10-08 10:45:17 +0000
@@ -3776,7 +3776,7 @@
 AC_OUTPUT(Makefile lib-src/Makefile oldXMenu/Makefile \
 	doc/emacs/Makefile doc/misc/Makefile doc/lispintro/Makefile \
 	doc/lispref/Makefile src/Makefile \
-	lwlib/Makefile lisp/Makefile leim/Makefile, [
+	lwlib/Makefile lisp/Makefile leim/Makefile test/automated/Makefile, [
 
 ### Make the necessary directories, if they don't exist.
 for dir in etc lisp ; do
--- Makefile.in	2010-06-12 03:35:55 +0000
+++ Makefile.in	2010-10-08 10:57:40 +0000
@@ -135,7 +135,7 @@
 infodir=@infodir@
 INFO_FILES=ada-mode auth autotype calc ccmode cl dbus dired-x ebrowse	\
            ede ediff edt eieio efaq eintr elisp emacs emacs-mime epa erc \
-	   eshell eudc flymake forms gnus idlwave info mairix-el	\
+	   ert eshell eudc flymake forms gnus idlwave info mairix-el	\
 	   message mh-e newsticker nxml-mode org pcl-cvs pgg rcirc	\
 	   reftex remember sasl sc semantic ses sieve smtpmail speedbar \
 	   tramp url vip viper widget woman
@@ -265,7 +265,7 @@
 SUBDIR = lib-src src lisp
 
 # The subdir makefiles created by config.status.
-SUBDIR_MAKEFILES = lib-src/Makefile doc/emacs/Makefile doc/misc/Makefile doc/lispref/Makefile doc/lispintro/Makefile src/Makefile oldXMenu/Makefile lwlib/Makefile leim/Makefile lisp/Makefile
+SUBDIR_MAKEFILES = lib-src/Makefile doc/emacs/Makefile doc/misc/Makefile doc/lispref/Makefile doc/lispintro/Makefile src/Makefile oldXMenu/Makefile lwlib/Makefile leim/Makefile lisp/Makefile test/automated/Makefile
 
 # Subdirectories to install, and where they'll go.
 # lib-src's makefile knows how to install it, so we don't do that here.
@@ -366,7 +366,8 @@
           $(srcdir)/oldXMenu/Makefile.in \
           $(srcdir)/lwlib/Makefile.in \
           $(srcdir)/leim/Makefile.in \
-          $(srcdir)/lisp/Makefile.in
+          $(srcdir)/lisp/Makefile.in \
+          $(srcdir)/test/automated/Makefile.in
 	./config.status
 
 config.status: ${srcdir}/configure ${srcdir}/lisp/version.el
@@ -817,7 +818,7 @@
 	cd src; $(MAKE) tags
 
 check:
-	@echo "We don't have any tests for GNU Emacs yet."
+	cd test/automated; $(MAKE) check
 
 dist:
 	cd ${srcdir}; ./make-dist
--- etc/NEWS	2010-10-08 03:17:31 +0000
+++ etc/NEWS	2010-10-10 12:06:45 +0000
@@ -195,6 +195,10 @@
 `package-enable-at-startup' to nil.  To change which packages are
 loaded, customize `package-load-list'.
 
+** An Emacs Lisp testing framework is now included.
+Emacs Lisp developers can use this to write automated tests for their
+code.  See the ERT info manual for details.
+
 ** The user option `remote-file-name-inhibit-cache' controls whether
 the remote file-name cache is used for read access.
 
--- ChangeLog	2010-10-09 18:46:57 +0000
+++ ChangeLog	2010-10-10 11:53:42 +0000
@@ -0,0 +1,9 @@
+2010-10-10  Christian Ohler  <ohler+emacs@fastmail.net>
+
+	* Makefile.in (INFO_FILES): Add ERT.
+
+	* Makefile.in (check): Run tests in test/automated.
+
+	* Makefile.in:
+	* configure.in: Add test/automated/Makefile.
+
--- doc/misc/ChangeLog	2010-10-10 00:15:21 +0000
+++ doc/misc/ChangeLog	2010-10-10 11:49:04 +0000
@@ -0,0 +1,7 @@
+2010-10-10  Christian Ohler  <ohler+emacs@fastmail.net>
+
+	* ert.texi: New file.
+
+	* Makefile.in:
+	* makefile.w32-in: Add ert.texi.
+
=== modified file 'etc/ChangeLog'
--- etc/ChangeLog	2010-09-23 19:00:31 +0000
+++ etc/ChangeLog	2010-10-10 11:55:10 +0000
@@ -0,0 +1,4 @@
+2010-10-10  Christian Ohler  <ohler+emacs@fastmail.net>
+
+	* NEWS: Mention ERT.
+
--- lisp/ChangeLog	2010-10-10 03:40:58 +0000
+++ lisp/ChangeLog	2010-10-10 11:58:24 +0000
@@ -0,0 +1,6 @@
+2010-10-10  Christian Ohler  <ohler+emacs@fastmail.net>
+
+	* emacs/lisp/ert.el, emacs/lisp/ert-run.el:
+	* emacs/lisp/ert-batch.el, emacs/lisp/ert-ui.el:
+	* emacs/lisp/ert-x.el: New files.
+
--- test/ChangeLog	2010-09-20 14:57:01 +0000
+++ test/ChangeLog	2010-10-10 11:59:53 +0000
@@ -0,0 +1,8 @@
+2010-10-10  Christian Ohler  <ohler+emacs@fastmail.net>
+
+	* automated: New directory for automated tests.
+
+	* automated/ert-tests.el, automated/ert-x-tests.el: New files.
+
+	* automated/Makefile.in: New file.
+
